megamek/src/megamek/server/ScenarioLoader.java

```diff
@@ -623,100 +618,21 @@ private void parseAutoEject(Entity entity, String eject) {
-        if (camoGroup.equals("No Camo") || camoGroup.equals("None")) {
-            camoGroup = Camouflage.NO_CAMOUFLAGE;
-        } else if (camoGroup.equals("General")) {
-            camoGroup = AbstractIcon.ROOT_CATEGORY;
-        } else {
-            // If CamoGroup does not have a trailing slash, add one, since all
-            // subdirectories require it
-            if (camoGroup.charAt(camoGroup.length() - 1) != '/') {
-                camoGroup += "/"; //$NON-NLS-1$
-            }
-        }
-        
-        boolean validGroup = false;
-
-        if (Camouflage.NO_CAMOUFLAGE.equals(camoGroup) || AbstractIcon.ROOT_CATEGORY.equals(camoGroup)) {
-            validGroup = true;
-        } else {
-            Iterator<String> catNames = MMStaticDirectoryManager.getCamouflage().getCategoryNames();
-            while (catNames.hasNext()) {
-                String s = catNames.next();
-                if (s.equals(camoGroup)) {
-                    validGroup = true;
-                }
-            }
-        }
-
-        return validGroup ? camoGroup : null;
-    }
-    
-    private String getValidCamoName(String camoGroup, String camoName) {
-        boolean validName = false;
-
-        // Validate CamoName
-        if (Camouflage.NO_CAMOUFLAGE.equals(camoGroup) && !StringUtil.isNullOrEmpty(camoGroup)) {
-            return PlayerColour.parseFromString(camoName).name();
-        } else {
-            Iterator<String> camoNames;
-            if (AbstractIcon.ROOT_CATEGORY.equals(camoGroup)) {
-                camoNames = MMStaticDirectoryManager.getCamouflage().getItemNames("");
-            } else {
-                camoNames = MMStaticDirectoryManager.getCamouflage().getItemNames(camoGroup);
-            }
-            while (camoNames.hasNext()) {
-                String s = camoNames.next();
-                if (s.equals(camoName)) {
-                    validName = true;
-                }
-            }
-        }
-        
-        return validName ? camoName : null;
-    }
     
     /*
      * Camo Parser/Validator for Individual Entity Camo
      */
-    private void parseCamo(Entity entity, String camoString) throws ScenarioLoaderException {
+    private void parseCamo(Entity entity, String camoString) {
         String[] camoData = camoString.split(SEPARATOR_COMMA, -1);
-        String camoGroup = getValidCamoGroup(camoData[0]);
-        if (camoGroup == null) {
-            throw new ScenarioLoaderException("invalidIndividualCamoGroup",
-                camoData[0], entity.getDisplayName());
-        }
-        String camoName = getValidCamoName(camoGroup, camoData[1]);
-        if (camoName == null) {
-            throw new ScenarioLoaderException("invalidIndividualCamoName",
-                camoData[1], camoGroup, entity.getDisplayName());
-        }
-
-        entity.setCamoCategory(camoGroup);
-        entity.setCamoFileName(camoName);
+        entity.setCamouflage(new Camouflage(camoData[0], camoData[1]));
```

Comment:
How does the camouflage constructor react to being passed bad data for either of those parameters?

Also, let's be a little more defensive here about the size of the array resulting from the split.

Reply:
If it returns null, it reverts to defaults. Otherwise, that's handled during camouflage creation by a fallback.

Reply:
Pushed up a better way to handle this, and merged the two methods into a single safer method.